Code/GraphMol/DistGeomHelpers/BoundsMatrixBuilder.cpp

```diff
@@ -72,8 +72,10 @@ class ComputedData {
     bondAdj.reset(bAdj);
     auto *bAngles = new RDNumeric::DoubleSymmMatrix(nBonds, -1.0);
     bondAngles.reset(bAngles);
-    cisPaths.resize(nBonds * nBonds * nBonds);
-    transPaths.resize(nBonds * nBonds * nBonds);
+    unsigned long capacity =
+        static_cast<unsigned long>(nBonds) * nBonds * nBonds;
+    cisPaths.resize(capacity);
```

Comment:
The final target for resize is (BITSET::size_type), 

```
BITSET::resize(size_type)
```

perhaps we could use this over unsigned long here.  Otherwise if you wanted to add a check you could do:

```
BITSET::size_type capacity = boost::numeric_cast<BITSET::size_type>( 
    static_cast<unsigned long>(nBonds) * nBonds * nBonds );
```

I don't think this really matters though as we'll run out of memory long before the value overflows.

Reply:
Using that form totally makes sense. It should theoretically also catch problems if the code is somehow run on a system where BITSET::size_type is too small.